services/ymax-planner/src/plan-deposit.ts

```diff
@@ -115,37 +116,35 @@ const computeWeightedTargets = (
   brand: Brand<'nat'>,
   current: Partial<Record<AssetPlaceRef, NatAmount>>,
   delta: bigint,
-  allocation: Record<PoolKey, number>,
+  allocation: TargetAllocation,
 ): Partial<Record<AssetPlaceRef, NatAmount>> => {
   const totalCurrentAmt = Object.keys(current).reduce(
     (acc, k) => (allocation[k] ? AmountMath.add(acc, current[k]) : acc),
     AmountMath.makeEmpty(brand),
   );
   const total = totalCurrentAmt.value + delta;
   assert(total >= 0n, X`total after delta must not be negative`);
-  const entries = Object.entries(allocation) as Array<
-    [PoolKey, number | bigint]
-  >;
+  const entries = Object.entries(allocation) as Array<[PoolKey, NatValue]>;
```

Comment:
`import { typedEntries } from '@agoric/internal';` should eliminate the need for such casts.
```suggestion
  const entries = typedEntries(allocation);
```

Reply:
typedEntries cannot quite handle it; it thinks the entries could be undefined.